extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1106,6 +1113,47 @@ export class Project implements ISqlProject {
 
 		return firstPropertyElement.childNodes[0].data;
 	}
+
+	/**
+	 * Adds folder items for all intermediate folders in the path.
+	 * For example, when adding a file `a/b/c/d.sql`, we also need to add items for
+	 * the following folders:
+	 *   a/
+	 *   a/b/
+	 *   a/b/c/
+	 *
+	 * @param absolutePath Normalized absolute path to the file or folder.
+	 */
+	private async ensureFolderItemsForPath(absolutePath: string): Promise<void> {
+		const normalizedProjectFolderPath = path.normalize(this.projectFolderPath);
```

Comment:
Here I think it would be nice if we could normalize the path in the constructor, where it is set, but I was not sure if it will break anything else somewhere, so I kept the logic here. I'm not super happy with it though.